### DISCUSSION TITLE
build with pyarmor: incorrectly assumes pure wheels

### DISCUSSION THREAD

**Discussion by nschloe** [pypa/cibuildwheel#965] [Q&A] [answered]
I have a pure Python module that I would like to publish using [pyarmor](https://pyarmor.readthedocs.io/en/latest/platforms.html)'s new `pyproject.toml` build backend.
```toml
[build-system]
requires = [
  "setuptools>=59",
  "wheel",
  "pyarmor>=7.2.2"
]
build-backend = "pyarmor.build_meta"
```
Unfortunately, `cibuildwheels` and also pypa-build,
```
python3 -m build --wheel
```
only see the pure Python project and a seemingly platform-indendent wheel is build (`foobar-1.0.0-py3-none-any.whl`). cibuildwheel errors out. Any idea if I or `pyarmor.build_meta` can enforce platform-dependent builds?

I tried being a wise guy by adding a dummy `setuptools.Extension` in a `setup.py`:
```python
from setuptools import Extension, setup

if __name__ == "__main__":
    setup(
        ext_modules=[Extension(name="dummy.module", sources=[])],
        zip_safe=False,
    )
```
This actually works locally with `python3 -m build --wheel`, but cibuildwheel still error out
with
```
gcc: fatal error: no input files
```
I guess it chokes on the fact that there are no sources. This could be a bug.

**Comment by henryiii** [comment, marked as answer]
I think this is an issue with pyarmor. Does it normally build platform dependent wheels? As far as I can tell, you don't want Python and platform dependent wheels, but only platform dependent wheels? I think you can meddle with the setuptools options for that. Here's an example of meddling with the platform tag: https://github.com/scikit-build/ninja-python-distributions/pull/85/files (not quite the same, but hopefully enough to get you started). You can use build as the backend, by the way:

```toml
[tool.cibuildwheel] build-frontend = "build"
```

Otherwise it just uses `pip wheel`.

Though I think, since you are not building anything specific to platforms other than... Well, I'm not sure what, then you'd just need build; you don't need manylinux, etc. I think?

**Reply by nschloe** [reply]
Thanks for the reply! pyarmor have [renamed their wheels](https://github.com/dashingsoft/pyarmor/commit/c8cf9ea48e32065cad3c26fa7b2d80107168a4c4) and everything works now.
